concent_api/core/models.py

```diff
@@ -73,3 +79,252 @@ class MessageAuth(Model):
     message              = OneToOneField(StoredMessage, related_name = 'auth')
     provider_public_key  = Base64Field(max_length=64)
     requestor_public_key = Base64Field(max_length=64)
+
+
+class ClientManager(Manager):
+
+    def get_or_create_full_clean(self, public_key: bytes):
+        """
+        Returns Model instance.
+        Does the same as get_or_create method, but also performs full_clean() on newly created instance.
+        """
+        try:
+            instance = self.get(public_key = base64.b64encode(public_key))
+        except self.model.DoesNotExist:
+            instance = self.model(
+                public_key_bytes = public_key
+            )
+            instance.full_clean()
+            instance.save()
+        return instance
+
+
+class Client(Model):
+    """
+    Represents Concent client, identified by public key.
+    """
+
+    objects = ClientManager()
+
+    public_key = Base64Field(max_length = CLIENT_ETHEREUM_KEY_LENGTH, unique = True)
+
+
+class Subtask(Model):
+    """
+    Represents subtask states.
+    """
+
+    class SubtaskState(ChoiceEnum):
```

Comment:
It's probably too late to point it out now, but it struck me just now that it could be `Subtask.State` rather than `Subtask.SubtaskState`.

Reply:
I would like not change it now, because of possible conflicts while rebasing on upper branches, if you want we can create separate issue out of it.

Reply:
Yeah. Like I said, it's probably too late now. I won't be creating a separate issue because it's not all that important but if you like the idea consider doing it as a small piece or refactoring at some point in the future.